tpls/desul/include/desul/atomics/Fetch_Op_HIP.hpp

```diff
@@ -22,6 +22,12 @@ namespace Impl {
                                    val,                                 \
                                    HIPMemoryOrder<MemoryOrder>::value,  \
                                    HIPMemoryScope<MemoryScope>::value); \
+  }                                                                     \
+  template <class MemoryOrder, class MemoryScope>                       \
+  __device__ inline T device_atomic_##OP##_fetch(                       \
+      T* ptr, T val, MemoryOrder order, MemoryScope scope) {            \
+    return OP##_fetch_operator<T, T>::apply(                            \
+        device_atomic_fetch_##OP(ptr, val, order, scope), val);         \
```

Comment:
It's kind of strange that the `FETCH_OP` macros instantiate both `fetch_OP` and `OP_fetch` . I would rename the macros to avoid any confusion.